app.js

```diff
@@ -5,29 +5,29 @@ var config = require('./config');
 
 // -- Configure JavaScript Runtime ---------------------------------------------
 
-var hasNativeIntl    = !!global.Intl;
 var hasNativePromise = !!global.Promise;
 
 require('es6-shim');
 
 // es6-shim provides a Promise implementation, but it's not very good.
-hasNativePromise || (global.Promise = require('promise'));
+if (!hasNativePromise) {
+    global.Promise = require('promise')
+}
 
-// Determine if the native `Intl` has the locale data we need.
-var hasNativeLocaleData = hasNativeIntl &&
-    config.availableLocales.every(function (locale) {
-        return Intl.NumberFormat.supportedLocalesOf(locale)[0] === locale;
-    });
+var areIntlLocalesSupported = require('intl-locales-supported');
 
-if (!hasNativeIntl) {
+if (global.Intl) {
+    // Determine if the built-in `Intl` has the locale data we need.
+    if (!areIntlLocalesSupported(config.availableLocales)) {
+        // `Intl` exists, but it doesn't have the data we need, so load the
+        // polyfill and replace the constructors with need with the polyfill's.
+        require('intl');
```

Comment:
Oh wow, I know this has nothing to do with this PR, but that's a pattern I don't like because it is not clear what happens here. I would much prefer writing something like `require('intl').polyfill();`...